File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.swift
import Foundation

func solution(_ n:Int) -> Int {
    
    let cnt = n.nonzeroBitCount
    
    var answer = n+1
    
    while true {
        if answer.nonzeroBitCount == cnt {
            return answer
        } else {
            answer += 1
        }
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def numToBinary(n):
    bin = ""
    
    while n > 0:
        bin += str(n % 2)
        n = int(n / 2)
    
    return bin[::-1]

def solution(n):
    
    bin = numToBinary(n)
    oneCnt = bin.count("1")
    
    answer = n+1
    
    while True:
        nextBin = numToBinary(answer)
        if nextBin.count("1") == oneCnt:
            return answer
        answer += 1